src/Domain/Direction.hpp

```diff
@@ -0,0 +1,158 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines class template Direction.
+
+#pragma once
+
+#include <functional>
+#include <iosfwd>
+#include <pup.h>
+
+#include "Domain/Side.hpp"
+#include "Parallel/PupStlCpp11.hpp"
+
+/// \ingroup ComputationalDomain
+/// A particular Side along a particular coordinate Axis.
+template <size_t VolumeDim>
+class Direction {
+ public:
+  /// The logical-coordinate names of each dimension
+  enum class Axis;
+
+  /// Construct by specifying an Axis and a Side.
+  Direction(Axis axis, Side side) : axis_(axis), side_(side) {}
+
+  /// Construct by specifying a dimension and a Side.
+  Direction<VolumeDim>(size_t dimension, Side side);
+
+  /// Default constructor for Charm++ serialization.
+  Direction() = default;
+
+  /// The dimension of the Direction
+  size_t dimension() const { return static_cast<size_t>(axis_); }
+
+  /// The Axis of the Direction
+  Axis axis() const { return axis_; }
+
+  /// The side of the Direction
+  Side side() const { return side_; }
+
+  /// The sign for the normal to the Side.
+  double sign() const { return (Side::Lower == side_ ? -1.0 : 1.0); }
+
+  /// The opposite Direction.
+  Direction<VolumeDim> opposite() const;
+
+  // {@
+  /// Helper functions for creating specific Directions.
+  /// These are labeled by the logical-coordinate names (Xi,Eta,Zeta).
+  // Note: these are functions because they contain static_assert.
+  static Direction<VolumeDim> lower_xi();
+  static Direction<VolumeDim> upper_xi();
+  static Direction<VolumeDim> lower_eta();
+  static Direction<VolumeDim> upper_eta();
+  static Direction<VolumeDim> lower_zeta();
+  static Direction<VolumeDim> upper_zeta();
+  // @}
+
+  /// Serialization for Charm++
+  void pup(PUP::er& p);  // NOLINT
+
+ private:
+  Axis axis_{Axis::Xi};
+  Side side_{Side::Lower};
+};
+
+/// Output operator for a Direction.
+template <size_t VolumeDim>
+std::ostream& operator<<(std::ostream& os,
+                         const Direction<VolumeDim>& direction);
+
+//##############################################################################
+// INLINE DEFINITIONS
+//##############################################################################
+
+// Needed in order to address warning:
```

Comment:
Should add "Ignorning `-Wpedantic` is needed" in case the comment gets lost from the code at some point